src/key/key.js

```diff
@@ -170,13 +170,13 @@ class Key {
   }
 
   /**
-   * Clones the key object
-   * @param {Boolean} [deep=false] Whether to return a deep clone
+   * Clones the key object. The copy is shallow, as it references the same packet objects as the original. However, if the top-level API is used, the two key instances are effectively independent.
+   * @param {Boolean} [forKeyEncryption=false] Only relevant for private keys: whether `encrypt()` is to be called either on the clone or the original, in which case the secret key paramenters are deeply copied.
    * @returns {Promise<Key>} Clone of the key.
    */
-  clone(deep = false) {
+  clone(forKeyEncryption = false) {
     const key = new this.constructor(this.toPacketList());
-    if (deep) {
+    if (forKeyEncryption) {
```

Comment:
The thing is, it literally does what is needed to encrypt the clone 😅 but yeah the most tangible thing would be that it clones the private params.
I just wonder whether then people will ask/wonder e.g. "why are not the public params cloneable" and similar.

Reply:
Sure, I mean, you can explain in a comment why we (might want to) clone the private params, but still I think it will be much more clear if you make the param name describe what it does :)